src/Game.hs

```diff
@@ -0,0 +1,243 @@
+{-# LANGUAGE InstanceSigs #-}
+{-# OPTIONS_GHC -Wno-unrecognised-pragmas #-}
+module Game where
+import Data.Array.IO
+
+import System.Random
+import Control.Monad
+import Control.Monad.State
+import Data.List
+import Data.List.Split
+import Prelude hiding (round)
+
+data Suit = Spades | Hearts | Clubs | Diamonds deriving (Eq,Show,Enum,Read)
+data Value = Two | Three | Four | Five | Six | Seven | Eight | Nine | Ten | Joker | Queen | King | Ace deriving (Ord,Eq,Show,Enum,Read)
+data Card = Card {suit:: Suit, value::Value} deriving (Eq,Show,Read)
+
+
+class Named a where
+    name :: a -> String
+
+
+data Player = Player {_nameP::String, cards::[Card]} deriving (Eq,Show,Read)
+data Team = Team {_nameT::String, points::Int, players::(String,String)} deriving (Eq,Show,Read)
```

Comment:
Should players be tuple of `Player` rather than string?

Reply:
I did that but we have to update the cards at two places then after i change the state , so i changed that to strings to reduce changes multiple places

Reply:
Got it, looks good 👍